database/src/main/java/com/firebase/ui/database/FirebaseArray.java

```diff
@@ -41,15 +41,15 @@ public FirebaseArray(Query ref) {
         mQuery.addChildEventListener(this);
```

Comment:
`public FirebaseArray(Query ref) {`

Leaving this public means anyone can create new `FirebaseArray`s which can lead to interesting usages such as #337.